resgroup: provide an API to get resgroup id of my proc.

pivotal-ning-yu opened the pull request:
Resource group id of my proc is stored in a local variable accessible
only in resgroup.c, but this information can also be interested in other
contexts, so an API `GetMyResGroupId()` is provided to get this
information.

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`

magi345 commented:
Thanks a lot!
@zhangteng5513 FYI

hlinnaka commented on src/backend/utils/resgroup/test/resgroup_test.c:
```diff
@@ -280,6 +280,19 @@ test_CpusetOperation(void **state)
+void
+test__GetMyResGroupId(void **state)
+{
+	/* self->groupId is initialized to InvalidOid */
```
This test seems pointless, the function being tested is a trivial getter function.

hlinnaka commented:
If the function isn't actually used for anything, why add it? Sooner or later, someone is just going to remove it again, since it's unused.

pivotal-ning-yu commented:
On Fri, Jan 11, 2019 at 5:04 PM Heikki Linnakangas <notifications@github.com>
wrote:

> If the function isn't actually used for anything, why add it? Sooner or
> later, someone is just going to remove it again, since it's unused.
>
This function is actually required by GPCC team, they will use it in their
code.  For the test part, yes I also think it is useless, I'd better remove
it.


> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/6681#issuecomment-453440295>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYjBHnKb7UvvupfaYc0eMbAcIL3wkRDlks5vCFO5gaJpZM4Z68iT>
> .
>

magi345 commented:
A similar discussion about unused code also happened in https://github.com/greenplum-db/gpdb/issues/6569
What's the best practice for maintaining such code?
Ideas come into my mind including,
 - Comment
 - Make a use case in `contrib`
Which can be our best practice?

pivotal-ning-yu commented:
Confirmed with GPCC team, there is a matrix collector extension (a shared library), it's loaded via `LD_PRELOAD` and will call this api.

@hlinnaka is there similar case in postgres?  Are there some functions that are called by extensions instead of postgres itself?  How are they defined?

In the other hand I'm thinking whether will it be better to convert this API to a UDF (or add a UDF wrapper for it) so people know its consumers can be outside of gpdb source code.

hlinnaka commented:
On 14/01/2019 05:11, Ning Yu wrote:
> Confirmed with GPCC team, there is a matrix collector extension (a
> shared library), it's loaded via `LD_PRELOAD` and will call this
> api.

LD_PRELOAD? That's sounds exotic.

> @hlinnaka is there similar case in postgres?  Are there some
> functions that are called by extensions instead of postgres itself?
> How are they defined?

I'm sure there are some. Comments usually go a long way.

- Heikki

pivotal-ning-yu commented:
On Mon, Jan 14, 2019 at 4:39 PM Heikki Linnakangas <notifications@github.com>
wrote:

> On 14/01/2019 05:11, Ning Yu wrote:
> > Confirmed with GPCC team, there is a matrix collector extension (a
> > shared library), it's loaded via `LD_PRELOAD` and will call this
> > api.
>
> LD_PRELOAD? That's sounds exotic.
>
> > @hlinnaka is there similar case in postgres? Are there some
> > functions that are called by extensions instead of postgres itself?
> > How are they defined?
>
> I'm sure there are some. Comments usually go a long way.
>

Thanks for the information.  I have updated the comments to explain about
the out-of-tree consumers.  Also removed the useless tests.


>
> - Heikki
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/greenplum-db/gpdb/pull/6681#issuecomment-453931226>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYjBHn_b98eyFc030Q6jZfNje8r7gt8wks5vDEIegaJpZM4Z68iT>
> .
>

pivotal-ning-yu commented:
Thanks for the review, merge it now.
